src/test/java/org/opensearch/neuralsearch/common/BaseNeuralSearchIT.java

```diff
@@ -733,11 +744,33 @@ protected Set<String> findDeployedModels() {
         List<Map<String, Object>> innerHitsMap = (List<Map<String, Object>>) hits.get("hits");
         return innerHitsMap.stream()
             .map(hit -> (Map<String, Object>) hit.get("_source"))
-            .filter(hitsMap -> !Objects.isNull(hitsMap) && hitsMap.containsKey("model_id"))
+            .filter(
+                hitsMap -> !Objects.isNull(hitsMap)
+                    && hitsMap.containsKey("model_id")
+                    && MLModelState.DEPLOYED.equals(getModelState(hitsMap.get("model_id").toString()))
```

Comment:
if no deployed models are found what is the expectation from the tests?

Reply:
If not deployed model was found, the test will fail in the assertion here: https://github.com/opensearch-project/neural-search/blob/main/src/test/java/org/opensearch/neuralsearch/common/BaseNeuralSearchIT.java#L748 
As I mentioned below, there seems to be a latency between task complete and `model_state` update to DEPLOYED. So I added a poll for model state there as well to avoid the issue.